File: AOJ/0384/0384.cpp
#include <iostream>
using namespace std;

long long getY(long long x){
  long long y = 0;
  while(x){
    y += x % 10;
    x /= 10;
  }
  return y;
}

int main(){
  long long a, n, m;
  cin >> a >> n >> m;
  int cnt = 0;
  for ( int y = 1; y <= 72; y++ ){
    long long x = 1;
    for ( int t = 1; t <= n; t++ ){
      x *= (y + a);
    }
    if (x <= m && getY(x) == y) cnt++;
  }
  cout << cnt << endl;
  return 0;
}

File: AOJ/0384/0384.py
def getY(x):
    y = 0
    while x > 0:
        y += x % 10
        x //= 10
    return y

a, n, m = map(int, input().split())
cnt = 0
for y in range(1, 73):
    x = 1
    for t in range(1, n + 1):
      x *= (y + a)
    if x <= m and getY(x) == y:
        cnt += 1

print(cnt)
